### PR TITLE
Fix guaranteed stack trace

### PR DESCRIPTION
**Pull request by ericdill**
I noticed that the original if/else block will always raise a stack trace. I'm not sure what the "ideal" behavior here is for logging, but the way it is currently implemented is a guaranteed NoneType has no attribute logger. 

@apetresc @aggFTW  do you think this naked `print` here is ok? Would you rather us work together to figure out how to get an active logger inject here so we can put this warning into the sparkmagic log file?

### CODE REVIEW

**Comment by Tagar** [comment]
cc @itamarst and @rgbkrk

**Comment by itamarst** [comment]
Could you include a test demonstrating the fix, or at least explain what this is trying to fix with an example traceback or reproducer? I'm not sure what this is trying to fix from skimming it.

File: sparkmagic/sparkmagic/livyclientlib/livysession.py [hunk 1]
```diff
@@ -17,38 +17,56 @@
...
+            # a billion iterations should be sufficient
+            run_at_most = int(1e9)
         self.run_at_most = run_at_most
 
     def run(self):
-        i = 0
-        if self.livy_session is not None:
-            self.livy_session.logger.info(u'Starting heartbeat for session {}'.format(self.livy_session.id))
-        else:
-            self.livy_session.logger.info(u'Will not start heartbeat because session is none')
```
Comment (ericdill) [inline sparkmagic/sparkmagic/livyclientlib/livysession.py]: Hi @itamarst. This is the offending line. On line 29, we are testing for `self.livy_session` being `None`. If the `else` bit of the clause gets executed, then `self.livy_session` is guaranteed to be `None` and so we will get a stack trace saying roughly "None type has no attribute 'logger'", which is a horribly cryptic and confusing error message. This PR fixes that issue, as well as adding some docs, giving better variable names and simplifying some looping logic.

Comment (itamarst) [reply sparkmagic/sparkmagic/livyclientlib/livysession.py]: Ah hah, makes sense.

**Comment by itamarst** [comment]
This definitely seems like an improvement, so going to merge it. Thank you!

**Pull request merged**